### PR TITLE
[CPDLP-3466] Create seed data on separation and sandbox

### PR DESCRIPTION
**Pull request by mooktakim**
### Context

- Ticket: https://dfedigital.atlassian.net.mcas.ms/browse/CPDLP-3466

### Changes proposed in this pull request

* New service to add completed mentors `ValidTestDataGenerator::CompletedMentorGenerator`
* Updated separation rake task to run the service

### CODE REVIEW

**Review by leandroalemao — APPROVED** [review]
Looks good tks Mook 👍 😄

**Review by ethax-ross — COMMENTED** [review]
I can't get this to run successfully; I run a `db:reset` then the rake task and it bombs out on a couple of different errors:

```
❯ rake lead_providers:seed_statements_and_applications
rake aborted!
ActiveRecord::RecordInvalid: Validation failed: School The school is already linked to this cohort (ActiveRecord::RecordInvalid)
```

```
 rake lead_providers:seed_statements_and_applications
rake aborted!
NoMethodError: undefined method `lead_provider' for nil:NilClass (NoMethodError)

      lead_provider: cpd_lead_provider.lead_provider,
```

File: lib/tasks/valid_test_data_generator/completed_mentor_generator.rb [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+    include ActiveSupport::Testing::TimeHelpers
+
+    class << self
+      def call(name:, cohort: Cohort.current, total_completed_mentors: 30)
+        new(name:, cohort:).call(total_completed_mentors:)
+      end
+    end
+
+    def call(total_completed_mentors:)
+      school = lead_provider.schools.order("RANDOM()").first
```
Comment (leandroalemao) [inline lib/tasks/valid_test_data_generator/completed_mentor_generator.rb:18]: `school = lead_provider.schools.joins(:school_cohorts).where(school_cohorts: { induction_programme_choice: "full_induction_programme" }).order("RANDOM()").first`

Think the generator is a bit flakey as I haven't got the same issue as Ross when I ran the first time.. but am getting it now.. that might fixes the issue I guess

Comment (mooktakim) [reply lib/tasks/valid_test_data_generator/completed_mentor_generator.rb]: Added the fix.👍

**Review by ethax-ross — APPROVED** [review]
Ran locally and working for me now 👍

**Pull request merged**